serde/src/java/org/apache/hadoop/hive/serde2/json/HiveJsonReader.java

```diff
@@ -397,7 +400,11 @@ private Object visitLeafNode(final JsonNode leafNode,
     case LONG:
       return Long.valueOf(leafNode.asLong());
     case BOOLEAN:
-      return Boolean.valueOf(leafNode.asBoolean());
+      if ("false".equalsIgnoreCase(leafNode.asText())) {
```

Comment:
why do we need a string match here, won't the `Boolean.valueOf(leafNode.asBoolean(true))` handle both cases